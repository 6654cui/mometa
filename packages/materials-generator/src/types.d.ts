export interface Material {
  name: string
  key: any
  assetGroups: AssetGroup[]
}
export interface AssetGroup {
  name: string
  key: any
  assets: Asset[]
}
export interface AssetImport {
  source: string
  mode: 'default' | 'named' | 'namespace'
  imported?: string
  local?: string
}
export interface Asset {
  name: string
  key: any
  cover?: string
  data: {
    code: string
    dependencies?: Record<string, AssetImport>
    sideEffectDependencies?: Array<string>
  }
}
