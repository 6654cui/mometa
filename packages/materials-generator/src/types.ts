export interface Material {
  name: string
  key: any
  assetGroups: AssetGroup[]
}

export interface AssetGroup {
  name: string
  key: any
  assets: Asset[]
}

// import { Button as AntButton } from 'antd'
export interface AssetImport {
  source: string // "antd"
  mode: 'default' | 'named' | 'namespace' // named
  imported?: string // "Button"
  local?: string // "AntButton"
}

export interface Asset {
  name: string
  key: any
  cover?: string

  data: {
    code: string // '<$ANT_BUTTON$></$ANT_BUTTON$>'
    dependencies?: Record<string, AssetImport> // { ANT_BUTTON: { name: 'Button', mode: 'named', source: 'antd' } }
    sideEffectDependencies?: Array<string>
  }
}
