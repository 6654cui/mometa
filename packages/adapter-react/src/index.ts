/** *
react 适配器，包括 editor compile / editor render
* @author
余聪
*/

export default function adapterReact() {}
