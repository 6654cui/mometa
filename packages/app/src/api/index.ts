import axiosStatic from 'axios'

const axios = axiosStatic.create({
  baseURL: 'https://jsonplaceholder.typicode.com/'
})

export async function getUsers() {
  const res = await axios.get('/users')
  return res.data
}
