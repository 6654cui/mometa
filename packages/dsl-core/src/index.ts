/**
 * DSL 的 parser &amp; stringify
 * @author 余聪
 */
import { ModuleNode } from './type'
import { parse, parseExpression } from '@babel/parser'

interface CommonOptions {
  plugins?: any[]
}
interface ParseOptions extends CommonOptions {
  id?: ModuleNode['id']
}
interface StringifyOptions extends CommonOptions {}

export function parseModuleNode(moduleCode: string, { id = null }: ParseOptions = {}): ModuleNode {
  return {
    type: 'module',
    id,
    imports: [],
    body: [],
    exports: []
  }
}

export function stringifyModuleNode(moduleNode: ModuleNode, opts?: StringifyOptions): { code: string } {
  return { code: '' }
}
