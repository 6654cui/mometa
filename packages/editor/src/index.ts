/** *
编辑器
* @author
余聪
*/

export default function editor() {}
