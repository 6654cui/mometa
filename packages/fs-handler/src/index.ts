/** *
 file updator
 * @author
 余聪
 */
import { waterFall } from 'run-seq'
import pify from 'pify'
import lazy from 'lazy-value'
import { createLineContentsByContent, LineContents, Range } from './utils/line-contents'
import { OpType } from './const'
export * from './const'

export { default as reactMiddlewares } from './react'

export interface CommonPreload extends Range {
  filename: string
  name: string
  text: string
}

export interface DelPreload extends CommonPreload {}

export interface RequestData {
  type: OpType.DEL
  preload: DelPreload
}

type Fs = Pick<typeof import('fs'), 'readFile' | 'writeFile'>

export interface MiddlewareContext {
  fs: Fs
  filename: string
  getContent: () => Promise<string>

  assertLocateByRange: () => Promise<{
    lineContents: LineContents
    lines: ReturnType<LineContents['locateByRange']>
  }>

  writeFile: (data: any) => Promise<void>
}

export type Middleware = (request: RequestData, ctx: MiddlewareContext, next: () => any) => any

const apiMiddle: Middleware = (data, ctx, next) => {
  ctx.assertLocateByRange = async () => {
    const content = await ctx.getContent()
    const lineContents = createLineContentsByContent(content, { filename: ctx.filename })
    const lines = lineContents.locateByRange(data.preload)

    const string = new LineContents(
      lines.map((x) => x.line),
      { filename: ctx.filename }
    ).toString(false)

    if (string !== data.preload.text) {
      throw new Error(`匹配错误
${JSON.stringify(string)} !== ${JSON.stringify(data.preload.text)}`)
    }
    return {
      lineContents,
      lines
    }
  }

  ctx.writeFile = async (data) => {
    return pify(ctx.fs.writeFile)(ctx.filename, data)
  }

  return next()
}

export function createFsHandler({ fs, middlewares = [] }: { fs: Fs; middlewares?: Middleware[] }) {
  return async (request: RequestData) => {
    // @ts-ignore
    const ctx: MiddlewareContext = {
      fs,
      filename: request.preload.filename,
      getContent: lazy(() => pify(fs.readFile)(request.preload.filename, 'utf8'))
    }

    return await waterFall([apiMiddle].concat(middlewares), [request, ctx])
  }
}
