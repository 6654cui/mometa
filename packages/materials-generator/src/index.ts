import Module = NodeJS.Module

export * from './utils/resolve-async-config'
export * from './utils/search-core'

export * from './plugins/create'

import { assets } from './plugins/assets'
import { groups } from './plugins/groups'
import { materials } from './plugins/materials'

export { assets, groups, materials }

// @ts-ignore
type SearchFn<T> = (dir: Parameters<T>[0]) => ReturnType<T>

export function createSearchHelper(mod: Module) {
  return {
    materials: ((dirs) => materials(dirs, mod.path)) as SearchFn<typeof materials>,
    groups: ((dirs) => groups(dirs, mod.path)) as SearchFn<typeof groups>,
    assets: ((dirs) => assets(dirs, mod.path)) as SearchFn<typeof assets>
  }
}
