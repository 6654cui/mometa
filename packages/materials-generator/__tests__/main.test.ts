import { materials, resolveLibMatConfig } from '../src'
import { fixture } from './helper'

describe('materialsGenerator', function () {
  it.skip('spec case', async function () {
    const x = await materials(fixture('deep'))
    expect(x).toMatchSnapshot()
  })

  it('resolveLibMatConfig', async function () {
    expect(await resolveLibMatConfig(fixture('mat-lib'))).toMatchInlineSnapshot(`
      Array [
        1,
        Object {
          "x": 1,
          "y": Array [
            1,
          ],
        },
      ]
    `)
  })

  it('resolveLibMatConfig invalid', async function () {
    await expect(resolveLibMatConfig(fixture('invalid-mat-lib'))).rejects.toMatchObject({
      _originalMessage: `Cannot find module '${fixture(
        'invalid-mat-lib',
        'mometa.js'
      )}' from 'src/utils/resolve-async-config.ts'`
    })
  })
})

// describe('reactComponentAsset', function () {
//   it('spec case', async function () {
//     // const x = await reactComponentAsset(fixture('my-comp/alert/index.tsx'))
//     // const x = await reactComponentAsset(fixture('antd/alert/index.d.ts'))
//     // expect(x).toMatchSnapshot()
//   })
// })
