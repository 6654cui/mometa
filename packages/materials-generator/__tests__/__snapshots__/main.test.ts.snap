// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`materialsGenerator spec case 1`] = `
Array [
  Object {
    "assetGroups": Array [
      Object {
        "key": "groups-a",
        "name": "groups-a",
      },
      Object {
        "assets": Array [],
        "key": "group-b",
        "name": "group-b",
      },
    ],
    "key": "a",
    "name": "a",
  },
]
`;
